docs/dev/README.md

```diff
@@ -0,0 +1,110 @@
+* [Compression]() - TODO: Write documentation for Compression component.
+* [Query Engine]() - TODO: Write documentation for Query Engine.
+* [Metadata]() - TODO: Write documentation for Metadata component.
+
+Beyond describing each individual component in detail we also provide a larger overview of the architecture of 
+ModelarDB, focusing on the integration between the above described components to accomplish the goals of the system.
+TODO: Add link to architecture documentation when new architecture is implemented and documented.
+
+## Development
+All code must be formatted according to the [Rust Style Guide](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md)
+using [rustfmt](https://github.com/rust-lang/rustfmt). Subjects not covered in the style guide, or requirements specific 
+to this repository, are covered here.
+
+### Documentation
+All public and private functions must have an accompanying doc comment that describes the purpose of the function. For 
+complex functions, the doc comment can also include a description of each parameter, the return value, and examples.
+
+All modules must have an accompanying doc comment that describes the general functionality of the module. A brief 
+description of the public functions, structs, enums, or other central elements of the module can be included.
+
+### Testing and Linting
+All public and private functions must be appropriately covered by unit tests. Full coverage is intended, which means all
+branches of computation within each function should be thoroughly tested.
+
+In addition, the following commands must not return any warnings or errors for the code currently in master: 
+[cargo build --all-targets](https://doc.rust-lang.org/cargo/commands/cargo-build.html), 
+[cargo clippy --all-targets](https://github.com/rust-lang/rust-clippy), 
+[cargo doc](https://doc.rust-lang.org/cargo/commands/cargo-doc.html), 
+[cargo machete --with-metadata](https://github.com/bnjbvr/cargo-machete), 
+and [cargo test --all-targets](https://doc.rust-lang.org/cargo/commands/cargo-test.html).
+
+### Crates
+To avoid confusion and unnecessary dependencies, a list of crates is included. Note that this only includes crates
+used for purposes such as logging, where multiple crates provide similar functionality.
+
+- Logging - [tracing](https://crates.io/crates/tracing)
+- Async Runtime - [tokio](https://crates.io/crates/tokio)
+- gRPC - [tonic](https://crates.io/crates/tonic)
+- UUID - [uuid](https://crates.io/crates/uuid)
+- SQLite - [rusqlite](https://crates.io/crates/rusqlite)
+- Memory Allocation - [snmalloc-rs](https://crates.io/crates/snmalloc-rs)
+- Hardware Information - [sysinfo](https://crates.io/crates/sysinfo)
+- Property-based Testing - [proptest](https://crates.io/crates/proptest)
+- Temporary Files and Directories - [tempfile](https://crates.io/crates/tempfile)
+
+## Docker
```

Comment:
Update the text to not mention development if the `Docker` documentation is moved to the user documentation.

Reply:
Even though the documentation is moved I think it makes sense to keep the sentence that mentions we do not recommend that it is used for active development. I also think it makes sense to keep the sentence about how it is ideal for developing other components that use `modelardbd` since that could still be considered using `modelardbd` and not developing `modelardbd`.